### DISCUSSION TITLE
Timestamps formatting

### DISCUSSION THREAD

**Discussion by marcustyphoon** [AprilSylph/XKit-Rewritten#108] [Suggestion Box]
I can easily imagine trying to please everyone with format options for timestamps getting way out of hand (been there, done that), and I'm certainly not suggesting trying that again :D

But I will note my personal preference, which is just-simple-relative-timestamps-on-the-bottom (i.e. swapping the text and the tooltips of the current behavior). I don't know what options might be preferred by a significant enough portion of users to be worth considering, but I figure you never know if you never bring it up!

**Comment by AprilSylph** [comment]
This is actually how I wanted Timestamps to work in the first place, but after discovering that `Intl.RelativeTimeFormat` doesn't do Twitter-esque super-short relative timestamps (e.g. "1d" instead of "1 day ago"), I decided to prioritise keeping the timestamp text as short as possible by only displaying the absolute timestamp.

I know on many posts there's enough space to fit extra text, but there's more to consider here - mainly mobile layout support, and masonry-style posts. In the future, there may also be more control buttons added by other features, even further reducing the spare horizontal space in the post footer. And while there *might* be solutions to these post footer real estate problems, there's no solution to blogs having long usernames with follow buttons appended, and I feel like decoupling the formatting of post timestamps and reblog timestamps isn't a route I want to go down.

That said, if `Intl.RelativeTimeFormat`'s `"narrow"` style ever evolves to be significantly more concise than its `"short"` style, I would implement this as the default behaviour immediately. As for it being a preference, I'll consider it if enough people want it badly enough, but for now I think it's best that the most unintrusive option is the only one.
